include/oneapi/mkl/dft/detail/descriptor_impl.hpp

```diff
@@ -46,6 +46,8 @@ inline commit_impl* get_commit(descriptor<prec, dom>& desc);
 template <precision prec, domain dom>
 class descriptor {
 public:
+    constexpr static auto precision = prec;
```

Comment:
I see what you are saying but it does help a lot here https://github.com/oneapi-src/oneMKL/pull/278/files#diff-7fb76b683c117a962220553f8ad016e187f45e5b5707936414d7662c1516ad30R68.

Do you have suggestions for an alternative? I could name it something weird to avoid collision

Reply:
Actually I guess I could just check the data-type. That should be easy.

Reply:
Alternatively you can make it private and make compute functions friends.

Reply:
I've just gone with my suggestion in https://github.com/oneapi-src/oneMKL/pull/278/commits/548a3d933e1501cbcac7135fec1beeb26724261f since it lead to quite simple code. I can reinvestigate this if it becomes an issue when doing the full implementation.